okhttp-dnsoverhttps/src/main/kotlin/okhttp3/dnsoverhttps/DnsOverHttps.kt

```diff
@@ -67,7 +69,24 @@ class DnsOverHttps internal constructor(
       }
     }
 
-    return lookupHttps(hostname)
+    return try {
+      lookupHttps(hostname)
+    } catch (bestFailure: UnknownHostException) {
+      val address = if (hostname.startsWith("[") && hostname.endsWith("]")) {
+        hostname.substring(1, hostname.length - 1)
+      } else {
+        hostname
+      }
+      if (address.canParseAsIpAddress()) {
```

Comment:
Nice find, I forgot this was a thing.  Can you invert this now?

Also if that function doesn't worry about [], we can probably skip that here also.

See https://github.com/square/okhttp/blob/3ad1912f783e108b3d0ad2c4a5b1b89b827e4db9/okhttp/src/jvmTest/java/okhttp3/HttpUrlTest.java#L514

Which confirms the HttpUrl strips the [].

Reply:
Do you mean that the HttpUrl will handle the square braces before calling dns lookup, so we don't need to check and strips the `[]` here and just remove the test case for hostname "[::1]"?